consensus/src/main/kotlin/maru/consensus/ProposerSelector.kt

```diff
@@ -15,10 +15,10 @@
  */
 package maru.consensus
 
-import maru.core.BeaconBlockHeader
 import maru.core.Validator
+import org.hyperledger.besu.consensus.common.bft.ConsensusRoundIdentifier
 import tech.pegasys.teku.infrastructure.async.SafeFuture
 
 interface ProposerSelector {
-  fun getProposerForBlock(header: BeaconBlockHeader): SafeFuture<Validator>
+  fun getProposerForBlock(consensusRoundIdentifier: ConsensusRoundIdentifier): SafeFuture<Validator>
```

Comment:
Be good to use the same method name as we do in `https://github.com/hyperledger/besu/blob/main/consensus/common/src/main/java/org/hyperledger/besu/consensus/common/bft/blockcreation/ProposerSelector.java` i.e. selectProposerForRound